credentials/src/main/java/me/uport/sdk/credentials/Credentials.kt

```diff
@@ -147,6 +149,38 @@ class Credentials(
     }
 
 
+    /**
+     * Authenticates [Selective Disclosure Response JWT](https://github.com/uport-project/specs/blob/develop/messages/shareresp.md) from uPort
+     * client as part of the [Selective Disclosure Flow](https://github.com/uport-project/specs/blob/develop/flows/selectivedisclosure.md).
+     *
+     * It Verifies and parses the given response token and verifies the challenge response flow.
+     *
+     * @param token **REQUIRED** a valid JWT response token
+     * @returns  a verified [JWTPayload]
+     * @throws [InvalidJWTException] when the challenge is failed or when the request token is unavailable
+     *
+     */
+    suspend fun authenticateDisclosure(token: String): JwtPayload {
+        val payload = JWTTools().verify(token)
+
+        if (payload.req == null) {
+            throw InvalidJWTException("Challenge was not included in response")
```

Comment:
@mirceanis I'm considering creating a new Exception type for these authentication errors. 
What are your thoughts?

Reply:
That's a good idea.
It should extend `InvalidJWTException`